### ISSUE TITLE
After upgrade from KOPS 1.8.0 (K8S 1.8.4) to 1.9.1 (K8S 1.9.9) nginx pods crashing in a loop

### ISSUE THREAD

**Issue by mitpwd** [kubernetes/kops#5489]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
```KOPS 1.9.1```

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
```Kubernetes 1.9.9```

**3. What cloud provider are you using?**
```AWS```

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
```
(fyi, rbac enabled on existing cluster)
kops edit cluster (updated K8S version from 1.8.4 to 1.9.9)
kops update cluster --yes
kops rolling-update cluster --yes
kubectl validate cluster (all OK)
kubectl create -f helm.yml
helm init --service-account helm
kubectl create serviceaccount --namespace kube-system tiller
kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
helm install -f nginx-values.yml --namespace nginx-ingress stable/nginx-ingress --name nginx-ingress
kubectl get pods -n nginx-ingress
```

**5. What happened after the commands executed?**
nginx pods were not starting anymore, they were in CrashLoopBackOff with restarts number growing. Tried version 0.14, 0.15 and latest 0.19. Same issue occurred.

Also checked the logs on the nginx pods, here is the result:
```
I0720 13:32:27.445290       7 flags.go:162] Watching for ingress class: nginx
W0720 13:32:27.445672       7 client_config.go:533] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0720 13:32:27.445835       7 main.go:158] Creating API client for https://*.*.*.*:443
```

**6. What did you expect to happen?**
nginx pods to start

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```
apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: 2018-04-20T10:34:55Z
  name: ******
spec:
  api:
    loadBalancer:
      type: Public
  authorization:
    rbac: {}
  channel: stable
  cloudLabels:
    Kops_Version: 1.9.1
    Kubectl_Version: 1.11.0
    Kubernetes_Version: 1.9.9
  cloudProvider: aws
  configBase: s3://******/******
  dnsZone: ******
  etcdClusters:
  - etcdMembers:
    - encryptedVolume: true
      instanceGroup: master-eu-west-2a
      name: a
    - encryptedVolume: true
      instanceGroup: master-eu-west-2b
      name: b
    - encryptedVolume: true
      instanceGroup: master-eu-west-2c
      name: c
    name: main
  - etcdMembers:
    - encryptedVolume: true
      instanceGroup: master-eu-west-2a
      name: a
    - encryptedVolume: true
      instanceGroup: master-eu-west-2b
      name: b
    - encryptedVolume: true
      instanceGroup: master-eu-west-2c
      name: c
    name: events
  fileAssets:
  - content: |
      apiVersion: audit.k8s.io/v1alpha1
      kind: Policy
      rules:
        # The following requests were manually identified as high-volume and low-risk,
        # so drop them.
        - level: None
          users: ["system:kube-proxy"]
          verbs: ["watch"]
          resources:
            - group: "" # core
              resources: ["endpoints", "services"]
        - level: None
          # Ingress controller reads `configmaps/ingress-uid` through the unsecured port.
          # TODO(#46983): Change this to the ingress controller service account.
          users: ["system:unsecured"]
          namespaces: ["kube-system"]
          verbs: ["get"]
          resources:
            - group: "" # core
              resources: ["configmaps"]
        - level: None
          users: ["kubelet"] # legacy kubelet identity
          verbs: ["get"]
          resources:
            - group: "" # core
              resources: ["nodes"]
        - level: None
          userGroups: ["system:nodes"]
          verbs: ["get"]
          resources:
            - group: "" # core
              resources: ["nodes"]
        - level: None
          users:
            - system:kube-controller-manager
            - system:kube-scheduler
            - system:serviceaccount:kube-system:endpoint-controller
          verbs: ["get", "update"]
          namespaces: ["kube-system"]
          resources:
            - group: "" # core
              resources: ["endpoints"]
        - level: None
          users: ["system:apiserver"]
          verbs: ["get"]
          resources:
            - group: "" # core
              resources: ["namespaces"]
        # Don't log these read-only URLs.
        - level: None
          nonResourceURLs:
            - /healthz*
            - /version
            - /swagger*
        # Don't log events requests.
        - level: None
          resources:
            - group: "" # core
              resources: ["events"]
        # Secrets, ConfigMaps, and TokenReviews can contain sensitive & binary data,
        # so only log at the Metadata level.
        - level: Metadata
          resources:
            - group: "" # core
              resources: ["secrets", "configmaps"]
            - group: authentication.k8s.io
              resources: ["tokenreviews"]
        # Get repsonses can be large; skip them.
        - level: Request
          verbs: ["get", "list", "watch"]
          resources:
            - group: "" # core
            - group: "admissionregistration.k8s.io"
            - group: "apps"
            - group: "authentication.k8s.io"
            - group: "authorization.k8s.io"
            - group: "autoscaling"
            - group: "batch"
            - group: "certificates.k8s.io"
            - group: "extensions"
            - group: "networking.k8s.io"
            - group: "policy"
            - group: "rbac.authorization.k8s.io"
            - group: "settings.k8s.io"
            - group: "storage.k8s.io"
        # Default level for known APIs
        - level: RequestResponse
          resources:
            - group: "" # core
            - group: "admissionregistration.k8s.io"
            - group: "apps"
            - group: "authentication.k8s.io"
            - group: "authorization.k8s.io"
            - group: "autoscaling"
            - group: "batch"
            - group: "certificates.k8s.io"
            - group: "extensions"
            - group: "networking.k8s.io"
            - group: "policy"
            - group: "rbac.authorization.k8s.io"
            - group: "settings.k8s.io"
            - group: "storage.k8s.io"
        # Default level for all other requests.
        - level: Metadata
    name: audit.yaml
    path: /srv/kubernetes/audit.yaml
    roles:
    - Master
  iam:
    allowContainerRegistry: true
    legacy: false
  kubeAPIServer:
    auditLogMaxAge: 10
    auditLogMaxBackups: 1
    auditLogMaxSize: 100
    auditLogPath: /var/log/kube-apiserver-audit.log
    auditPolicyFile: /srv/kubernetes/audit.yaml
    featureGates:
      AdvancedAuditing: "true"
  kubelet:
    featureGates:
      AdvancedAuditing: "true"
  kubernetesApiAccess:
  - *.*.*.*/32
  - *.*.*.*/32
  - *.*.*.*/32
  kubernetesVersion: 1.9.9
  masterInternalName: ******
  masterPublicName: api.******
  networkCIDR: *.*.*.*/16
  networkID: vpc-******
  networking:
    calico: {}
  nonMasqueradeCIDR: *.*.*.*/10
  sshAccess:
  - *.*.*.*/28
  - *.*.*.*/28
  subnets:
  - cidr: *.*.*.*/21
    name: eu-west-2a
    type: Private
    zone: eu-west-2a
  - cidr: *.*.*.*/21
    name: eu-west-2b
    type: Private
    zone: eu-west-2b
  - cidr: *.*.*.*/21
    name: eu-west-2c
    type: Private
    zone: eu-west-2c
  - cidr: *.*.*.*/24
    name: utility-eu-west-2a
    type: Utility
    zone: eu-west-2a
  - cidr: *.*.*.*/24
    name: utility-eu-west-2b
    type: Utility
    zone: eu-west-2b
  - cidr: *.*.*.*/24
    name: utility-eu-west-2c
    type: Utility
    zone: eu-west-2c
  topology:
    dns:
      type: Public
    masters: private
    nodes: private

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-04-20T10:34:55Z
  labels:
    kops.k8s.io/cluster: ******
  name: master-eu-west-2a
spec:
  image: ami-******
  machineType: t2.xlarge
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-west-2a
  role: Master
  subnets:
  - eu-west-2a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-04-20T10:34:55Z
  labels:
    kops.k8s.io/cluster: ******
  name: master-eu-west-2b
spec:
  image: ami-******
  machineType: t2.xlarge
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-west-2b
  role: Master
  subnets:
  - eu-west-2b

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-04-20T10:34:55Z
  labels:
    kops.k8s.io/cluster: ******
  name: master-eu-west-2c
spec:
  image: ami-******
  machineType: t2.xlarge
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-west-2c
  role: Master
  subnets:
  - eu-west-2c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-04-20T10:34:55Z
  labels:
    kops.k8s.io/cluster: ******
  name: nodes
spec:
  image: ami-*******
  machineType: t2.xlarge
  maxSize: 4
  minSize: 4
  nodeLabels:
    kops.k8s.io/instancegroup: nodes
  role: Node
  subnets:
  - eu-west-2a
  - eu-west-2b
  - eu-west-2c
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

```
kubectl get pods -n nginx-ingress -v 10
/.../
I0722 13:44:45.137298    4201 get.go:358] Unable to decode server response into a Table. Falling back to hardcoded types: attempt to decode non-Table object into a v1beta1.Table
I0722 13:44:45.137343    4201 get.go:358] Unable to decode server response into a Table. Falling back to hardcoded types: attempt to decode non-Table object into a v1beta1.Table
I0722 13:44:45.137362    4201 get.go:358] Unable to decode server response into a Table. Falling back to hardcoded types: attempt to decode non-Table object into a v1beta1.Table
I0722 13:44:45.137370    4201 get.go:358] Unable to decode server response into a Table. Falling back to hardcoded types: attempt to decode non-Table object into a v1beta1.Table
I0722 13:44:45.137377    4201 get.go:358] Unable to decode server response into a Table. Falling back to hardcoded types: attempt to decode non-Table object into a v1beta1.Table
NAME                                                        READY     STATUS             RESTARTS   AGE
doltish-rat-nginx-ingress-default-backend-cdfcccbbb-sls6s   1/1       Running            1          1d
doltish-rat-nginx-ingress-nginx-ingress-6kqf4               0/1       CrashLoopBackOff   954        1d
doltish-rat-nginx-ingress-nginx-ingress-btgnc               0/1       Running            950        1d
doltish-rat-nginx-ingress-nginx-ingress-cztm2               0/1       CrashLoopBackOff   923        1d
doltish-rat-nginx-ingress-nginx-ingress-pr54p               0/1       CrashLoopBackOff   927        1d`
```
 
```
kubectl describe pod doltish-rat-nginx-ingress-nginx-ingress-btgnc -n nginx-ingress -v 10

/.../
Name:           doltish-rat-nginx-ingress-nginx-ingress-btgnc
Namespace:      nginx-ingress
Node:           ip-*.*.*.*.eu-west-2.compute.internal/*.*.*.*
Start Time:     Fri, 20 Jul 2018 15:35:03 +0100
Labels:         app=nginx-ingress
                component=nginx-ingress
                controller-revision-hash=2311478611
                pod-template-generation=1
                release=doltish-rat
Annotations:    checksum/config=f674fae5f49736769e1693818957a7202b376b6251baf7ad7e685a0a33435f02
Status:         Running
IP:             *.*.*.*
Controlled By:  DaemonSet/doltish-rat-nginx-ingress-nginx-ingress
Containers:
  nginx-ingress-nginx-ingress:
    Container ID:  docker://85cff1d28a263c860453ae02f8bb4e6376d127e5f756086576a9d75886a08659
    Image:         ******/***/nginx:0.14.0
    Image ID:      docker-pullable://******/***/nginx@sha256:60710dd68e19c16c1c4ae0e8bd153c21fc4ce230e8b795493dd213e5c16e8eb9
    Ports:         80/TCP, 443/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      /nginx-ingress-controller
      --default-backend-service=nginx-ingress/doltish-rat-nginx-ingress-default-backend
      --election-id=ingress-controller-leader
      --ingress-class=nginx
      --configmap=nginx-ingress/doltish-rat-nginx-ingress-nginx-ingress
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    143
      Started:      Sun, 22 Jul 2018 13:44:38 +0100
      Finished:     Sun, 22 Jul 2018 13:44:58 +0100
    Ready:          False
    Restart Count:  950
    Liveness:       http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=3
    Readiness:      http-get http://:10254/healthz delay=10s timeout=1s period=10s #success=1 #failure=3
    Environment:
      POD_NAME:       doltish-rat-nginx-ingress-nginx-ingress-btgnc (v1:metadata.name)
      POD_NAMESPACE:  nginx-ingress (v1:metadata.namespace)
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from doltish-rat-nginx-ingress-token-gv6mj (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  doltish-rat-nginx-ingress-token-gv6mj:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  doltish-rat-nginx-ingress-token-gv6mj
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/disk-pressure:NoSchedule
                 node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
Events:
  Type     Reason     Age                   From                                                 Message
  ----     ------     ----                  ----                                                 -------
  Warning  Unhealthy  20m (x1873 over 1d)   kubelet, ip-*-*-*-*.eu-west-2.compute.internal  Liveness probe failed: Get http://*.*.*.*.3:10254/healthz: dial tcp *.*.*.*:10254: getsockopt: connection refused
  Warning  BackOff    30s (x11291 over 1d)  kubelet, ip-*-*-*-*.eu-west-2.compute.internal  Back-off restarting failed container
```

**9. Anything else do we need to know?**
```
anonymousAuth has been enabled, then disabled with a rolling upgrade every time.

  kubelet:
    anonymousAuth: false

  kubelet:
    anonymousAuth: true

```

**Comment by mapdegree** [comment]
It seems the DNS resolution does not work after the upgrade from within the Nginx pods.

**Comment by mapdegree** [comment]
Thank you. I do not have an issue when creating a cluster from scratch, only when upgrading from this specific version 1.8.4 to 1.9.x

**Comment by mapdegree** [comment]
We were able to fix this by first patching K8S to 1.8.12 and then upgrading KOPS in a separate step. We also had to leave the old AMI of CoreOS IDin place and let it update itself. We noticed journald did not indicate any root causes on the masters, just consequences that none of the containers could start. I consider this as closed without finding the root cause. Thanks.
